embedded-graphics/src/primitives/arc.rs

```diff
@@ -0,0 +1,632 @@
+            draw_above_a: (angle_start < ANGLE_90DEG) ^ negative_sweep,
+            draw_above_b: (angle_end >= ANGLE_90DEG) ^ negative_sweep,
+            sweep: arc.angle_sweep.abs(),
+        }
+    }
+
+    fn empty() -> Self {
+        Self {
+            line_a: LinearEquation::flat(),
+            line_b: LinearEquation::flat(),
+            draw_above_a: true,
+            draw_above_b: true,
+            sweep: Angle::zero(),
+        }
+    }
+}
+
+/// Iterator that returns only the points which are inside a plane sector defined by two lines.
+#[derive(Copy, Clone, PartialEq, PartialOrd, Debug)]
+struct PlaneSectorIterator {
+    plane_sector: PlaneSector,
+    points: super::rectangle::Points,
+}
+
+impl PlaneSectorIterator {
+    fn new(arc: &Arc) -> Self {
+        Self {
+            plane_sector: PlaneSector::new(arc),
+            points: arc.bounding_box().points(),
+        }
+    }
+
+    fn empty() -> Self {
+        Self {
+            plane_sector: PlaneSector::empty(),
+            points: Rectangle::new(Point::zero(), Size::zero()).points(),
+        }
+    }
+}
+
+impl Iterator for PlaneSectorIterator {
+    type Item = Point;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let plane_sector = self.plane_sector;
+        self.points.find(|p| {
+            let side_a = plane_sector.line_a.side(p);
+            let side_b = plane_sector.line_b.side(p);
+
+            let correct_a_side = plane_sector.draw_above_a ^ (side_a == LineSide::BELLOW);
+            let correct_b_side = plane_sector.draw_above_b ^ (side_b == LineSide::BELLOW);
+
+            if plane_sector.sweep < ANGLE_180DEG {
+                correct_a_side && correct_b_side
+            } else if plane_sector.sweep < ANGLE_360DEG {
+                correct_a_side || correct_b_side
+            } else {
+                true
+            }
+        })
+    }
+}
+
+/// Iterator that returns the squared distance to the center for all points in the bounding box.
+#[derive(Copy, Clone, PartialEq, PartialOrd, Debug)]
+struct DistanceIterator {
+    center: Point,
+    points: PlaneSectorIterator,
+}
+
+impl DistanceIterator {
+    fn new(arc: &Arc) -> Self {
+        Self {
+            center: arc.center_2x(),
+            points: PlaneSectorIterator::new(arc),
+        }
+    }
+
+    fn empty() -> Self {
+        Self {
+            center: Point::zero(),
+            points: PlaneSectorIterator::empty(),
+        }
+    }
+}
+
+impl Iterator for DistanceIterator {
+    type Item = (Point, u32);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.points.next().map(|p| {
+            let delta = self.center - p * 2;
+            let distance = delta.x.pow(2) as u32 + delta.y.pow(2) as u32;
+            (p, distance)
+        })
+    }
+}
+
+/// Define one side of a line
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+enum LineSide {
+    ABOVE,
+    BELLOW,
+}
```

Comment:
Enum variants need to be camel case and there is a typo:
```suggestion
enum LineSide {
    Above,
    Below,
}
```